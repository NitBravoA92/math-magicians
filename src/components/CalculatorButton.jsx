import PropTypes from 'prop-types';

const NumberButton = ({ name, classes, interactivity }) => (<button type="button" onClick={ () => interactivity(name) } className={`btn ${classes.join(' ')}`}>{ name }</button>);

NumberButton.propTypes = {
  name: PropTypes.string.isRequired,
  classes: PropTypes.arrayOf(PropTypes.string).isRequired,
  interactivity: PropTypes.func.isRequired,
};

export default NumberButton;
