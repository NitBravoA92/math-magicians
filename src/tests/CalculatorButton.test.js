import React from 'react';
import '@testing-library/jest-dom/extend-expect';
import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';
import CalculatorButton from '../components/CalculatorButton';

describe('The CalculatorButton component', () => {
  test('renders correctly into the DOM', () => {
    const operations = (name) => name;
    const name = '5';

    const component = renderer
      .create(<CalculatorButton name={name} classes={['btn-gray']} interactivity={() => { operations(name); }} />)
      .toJSON();

    expect(component).toMatchSnapshot();
  });

  test('should render a button with the text: 1', () => {
    const buttonOne = render(<CalculatorButton name="1" classes={['btn-gray']} interactivity={() => {}} />);

    expect(buttonOne.container).toHaveTextContent('1');
  });

  test('should render a button with the text: รท', () => {
    const buttonDivider = render(<CalculatorButton name="รท" classes={['btn-orange']} interactivity={() => {}} />);

    expect(buttonDivider.container).toHaveTextContent('รท');
  });

  test('should render a button with the text: AC', () => {
    const buttonAC = render(<CalculatorButton name="AC" classes={['btn-gray']} interactivity={() => {}} />);

    expect(buttonAC.container).toHaveTextContent('AC');
  });

  test('should render a button with the text: 9', () => {
    const buttonNine = render(<CalculatorButton name="9" classes={['btn-gray']} interactivity={() => {}} />);

    expect(buttonNine.container).toHaveTextContent('9');
  });
});
