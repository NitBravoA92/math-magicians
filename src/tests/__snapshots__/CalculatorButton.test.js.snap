// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The CalculatorButton component renders correctly into the DOM 1`] = `
<button
  className="btn btn-gray"
  onClick={[Function]}
  type="button"
>
  5
</button>
`;
